File: steakfrye-cs50-2018-x-caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        //convert the key from command line into an integer
        string key = argv[1];
        int i = atoi(key);

        //prompt user for text desired to be encrypted
        string text = get_string("plaintext: ");

        printf("ciphertext: ");
        for (int j = 0, n = strlen(text); j < n; j++)
        {
            char x = text[j];
            char m;
            //check for upper or lower case and make the cipher
            if (isupper(x))
            {
                m = (x - 'A' + i) % 26;
                m = m + 'A';
                printf("%c", m);
            }
            else if (islower(x))
            {
                m = (x - 'a' + i) % 26;
                m = m + 'a';
                printf("%c", m);
            }
            //if it isn't a letter, go ahead and print as is
            else
            {
                printf("%c", x);
            }
        }
    }
    //If argc is not 2, print error
    else
    {
        printf("Error");
        return 1;
    }
    printf("\n");
    return 0;
}

File: pset6/caesar.py
#creates a simple cipher based on a user's input

import sys
from cs50 import get_string

#if user did not give a proper argument, error
if not len(sys.argv) == 2:
    print("Error")
    exit(1)

#get key and text from user
elif len(sys.argv) == 2:
    key = int(sys.argv[1])
    text = get_string("plaintext: ")

    #prepare to print cipher
    print("ciphertext: ", end="")
    for i in range(len(text)):
        ltr = text[i]

        while ltr.isupper():
            #convert letter to integer and wrap around alphabet
            cipher_ltr = (ord(ltr) - 65 + key) % 26
            cipher_ltr += 65 #return its case
            print(chr(cipher_ltr), end="") #print it back as a letter
            break

        while ltr.islower():
            cipher_ltr = (ord(ltr) - 97 + key) % 26
            cipher_ltr += 97
            print(chr(cipher_ltr), end="")
            break

        #if character is not a letter, print as is
        if not ltr.isalpha():
            print(ltr, end="")
    print()
    exit(0)